src/wazuh_modules/vulnerability_scanner/tests/unit/eventDetailsBuilder_test.cpp

```diff
@@ -1177,3 +1214,225 @@ TEST_F(EventDetailsBuilderTest, TestNoSuccessfulVulnerabilityStatus)
+    EXPECT_CALL(*spOsDataCacheMock, getOsData(_)).WillRepeatedly(testing::Return(osData));
+
+    spRemediationDataCacheMock = std::make_shared<MockRemediationDataCache>();
+    EXPECT_CALL(*spRemediationDataCacheMock, getRemediationData(_)).WillRepeatedly(testing::Return(Remediation {}));
+
+    m_spDatabaseFeedManagerMock = std::make_shared<MockDatabaseFeedManager>();
+    EXPECT_CALL(*m_spDatabaseFeedManagerMock, getVulnerabiltyDescriptiveInformation(_, _))
+        .WillRepeatedly(testing::Invoke(mockGetVulnerabiltyDescriptiveInformation));
+
+    ASSERT_TRUE(m_parser.Parse(syscollector_deltas_SCHEMA));
+    ASSERT_TRUE(m_parser.Parse(DELTA_PACKAGES_INSERTED_MSG.c_str()));
+    m_buffer = m_parser.builder_.GetBufferPointer();
+
+    std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*, const nlohmann::json*>
+        syscollectorDelta = SyscollectorDeltas::GetDelta(reinterpret_cast<const char*>(m_buffer));
+    m_scanContext = std::make_shared<TScanContext<TrampolineOsDataCache, GlobalData, TrampolineRemediationDataCache>>(
+        syscollectorDelta);
+    // Mock one vulnerability
+    m_scanContext->m_elements[CVEID] =
+        R"({"operation":"INSERTED", "id":"001_ec465b7eb5fa011a336e95614072e4c7f1a65a53_CVE-2024-1234"})"_json;
+
+    m_scanContext->m_vulnerabilitySource = std::make_pair(adp, adp);
+}
+
+void EventDetailsBuilderAdpTest::TearDown()
+{
+    spOsDataCacheMock.reset();
+    spRemediationDataCacheMock.reset();
+    spGlobalDataMock.reset();
+    PolicyManager::instance().teardown();
+    std::filesystem::remove_all("queue/vd");
+}
+
+void EventDetailsBuilderAdpTest::TestingAdpDescription(const std::string& adp)
+{
+    spGlobalDataMock = std::make_shared<MockGlobalData>();
+    EXPECT_CALL(*spGlobalDataMock, vendorMaps()).WillRepeatedly(testing::Return(ADP_DESCRIPTIONS));
+
+    TEventDetailsBuilder<MockDatabaseFeedManager,
+                         TScanContext<TrampolineOsDataCache, GlobalData, TrampolineRemediationDataCache>,
+                         TrampolineGlobalData>
+        eventDetailsBuilder(m_spDatabaseFeedManagerMock);
+
+    EXPECT_NO_THROW(eventDetailsBuilder.handleRequest(m_scanContext));
+
+    auto& elementData = m_scanContext->m_elements[CVEID].at("data");
+
+    EXPECT_STREQ(elementData.at("vulnerability").at("scanner").at("source").get_ref<const std::string&>().c_str(),
+                 adp.c_str());
+}
+
+TEST_F(EventDetailsBuilderAdpTest, InvalidAdpTest)
+{
+    // Call custom setup function
+    const std::string adp {"Invalid"};
+    SetUp(adp);
+    TestingAdpDescription(" ");
+}
+
+TEST_F(EventDetailsBuilderAdpTest, NVDTest)
+{ // Call custom setup function
+    const std::string adp {"nvd"};
+    SetUp(adp);
+    TestingAdpDescription(ADP_DESCRIPTIONS.at("adp_descriptions").at("nvd").at("adp").get<std::string>().c_str());
+}
+
+TEST_F(EventDetailsBuilderAdpTest, CanonicalTest)
+{ // Call custom setup function
```

Comment:
Question: What do these comments mean? Can we remove them?
```suggestion
{
```